File: component/src/main/java/com/mackwu/component/jetpack/ViewModelActivity.java
package com.mackwu.component.jetpack;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import com.mackwu.component.R;
import com.mackwu.component.base.BaseActivity;

/**
 * ===================================================
 * Created by MackWu on 2020/6/30 11:48
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
public class ViewModelActivity extends BaseActivity {

    private UserViewModel userViewModel;

    @Override
    public int getLayoutId() {
        return R.layout.activity_test;
    }

    @Override
    public void initView(@Nullable final Bundle savedInstanceState) {
        userViewModel = new ViewModelProvider(this).get(UserViewModel.class);
    }

    @Override
    public void initData(@Nullable final Bundle savedInstanceState) {

    }

}

File: http/src/main/java/com/mackwu/http/jetpack/ViewModelActivity.kt
package com.mackwu.http.jetpack

import androidx.lifecycle.ViewModelProviders
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.mackwu.http.R
import kotlinx.android.synthetic.main.activity_main.*

/**
 * ===================================================
 * Created by MackWu on 2019/12/23 19:10
 * <a href="mailto:wumengjiao828@163.com">Contact me</a>
 * <a href="https://github.com/mackwu828">Follow me</a>
 * ===================================================
 */
class ViewModelActivity  : AppCompatActivity(){

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val ipViewModel = ViewModelProviders.of(this).get(IpViewModel::class.java)

        btn_test.setOnClickListener {  }
    }
}